crates/oxc_linter/src/rules/unicorn/number_literal_case.rs

```diff
@@ -0,0 +1,246 @@
+use oxc_ast::AstKind;
+use oxc_diagnostics::{
+    miette::{self, Diagnostic},
+    thiserror::{self, Error},
+};
+use oxc_macros::declare_oxc_lint;
+use oxc_span::Span;
+
+use crate::{context::LintContext, rule::Rule, AstNode, Fix};
+
+#[derive(Debug, Error, Diagnostic)]
+#[error("eslint-plugin-unicorn(number-literal-case): {1}")]
+#[diagnostic(severity(warning), help("{2}"))]
+struct NumberLiteralCaseDiagnostic(#[label] pub Span, &'static str, &'static str);
+
+#[derive(Debug, Default, Clone)]
+pub struct NumberLiteralCase;
+
+declare_oxc_lint!(
+    /// ### What it does
+    /// This rule enforces proper case for numeric literals.
+    ///
+    /// ### Why is this bad?
+    /// When both an identifier and a number literal are in lower case, it can be hard to differentiate between them.
+    ///
+    /// ### Example
+    /// ```javascript
+    /// // Fail
+    /// const foo = 0XFF;
+    /// const foo = 0xff;
+    /// const foo = 0Xff;
+    /// const foo = 0Xffn;
+    ///
+    /// const foo = 0B10;
+    /// const foo = 0B10n;
+    ///
+    /// const foo = 0O76;
+    /// const foo = 0O76n;
+    ///
+    /// const foo = 2E-5;
+    ///
+    /// // Pass
+    /// const foo = 0xFF;
+    /// const foo = 0b10;
+    /// const foo = 0o76;
+    /// const foo = 0xFFn;
+    /// const foo = 2e+5;
+    /// ```
+    NumberLiteralCase,
+    correctness
+);
+
+impl Rule for NumberLiteralCase {
+    fn run<'a>(&self, node: &AstNode<'a>, ctx: &LintContext<'a>) {
+        let (raw_literal, raw_span) = match node.kind() {
+            AstKind::NumberLiteral(number) => (number.raw, number.span),
+            AstKind::BigintLiteral(number) => {
+                let span = number.span;
+                (&ctx.source_text()[span.start as usize..span.end as usize], span)
```

Comment:
does this work?
```
(span.source_text(ctx.source_text(), span) 
```

Reply:
Good catch - updated in 2a58b6b77859ad28891c7036d007b834f93dee5f.